### DISCUSSION TITLE
增加 epub,azw,azw3,mobi,fb2,fb2.zip,cbz 等格式文件预览

### DISCUSSION THREAD

**Discussion by Febtw** [AlistGo/alist#3119] [Show and tell]
我有一个妄想，打开 Alist 就能浏览我的所有文件。
但我并不会创造工具，幸运的是我遇上了一些优秀的项目，不敢密敛，但愿连珠合璧。

## [johnfactotum/foliate-js](https://github.com/johnfactotum/foliate-js) --- Library for rendering e-books in the browser
支持 epub,azw,azw3,mobi,fb2,fb2.zip,cbz 格式
作者已增添api功能，[照此](https://github.com/johnfactotum/foliate-js/issues/3#issuecomment-1398034356)修改

**2023-01-21**
项目 demo 还未添加api，故需自行搭建使用
kfx/kf8 等格式没找到文件，故未进行测试
fb2/fb2.zip/cbz 的在线阅读功能还有问题，已报告作者

**过程：**
1、Nginx设置：参考[《AList添加epub文件预览功能流程分享》](https://github.com/alist-org/alist/discussions/2735)，新建目录为/foliatejs/
2、部署 foliate-js 文件，并[照此](https://github.com/johnfactotum/foliate-js/issues/3#issuecomment-1398034356)修改
3、reader.html 文件中，搜索 样式  .toolbar 将 width 改为 97% 避免前进按钮被全屏按钮遮挡
4、reader.js 文件中，搜索 menu.groups.layout.select('paginated')，将 paginated 改为 scrolled ，使默认滚动翻页
5、alist 管理 “Iframe 预览” 添加 
```
	"epub,epub3,azw,azw3,mobi,fb2,zip,cbz":{
	"foliate.js":"/foliatejs/reader.html?url=$durl"
	},
```
注意：zip是指 `.fb2.zip`，对于普通的zip文件不能打开要手动切换到 download

成果
![image](https://user-images.githubusercontent.com/96163355/213859319-dc07d533-6438-46d7-92d2-f211c2c42ab2.png)

**Comment by anwen-anyi** [comment]
弱弱的问下，我看到有 zip 这个zip是我认知当中的压缩包吗？

**Reply by Febtw** [reply]
> 弱弱的问下，我看到有 zip 这个zip是我认知当中的压缩包吗？

应该是zip打包的 fb2，原项目可以打开本地的 `.fb2.zip` 文件。我其实并不熟悉这种格式，api打开该格式还有问题，alist好像不能识别双后缀的扩展名，所以应该改成 zip，然后对于普通的zip文件要手动切换到 download
